charts/internal/azure-infra/templates/main.tf

```diff
@@ -73,6 +73,13 @@ resource "azurerm_network_security_group" "workers" {
   {{- end}}
 }
 
+{{ if .Values.identity.enabled -}}
```

Comment:
Nit: You could just use `if .Values.identity` to check this condition. Then you would "save" the `enabled` flag (it's not needed to have it explicitly as far as I can see). Same below.